### DISCUSSION TITLE
[@mantine/form] add a "submitting" property if "onSubmit" is a Promise.

### DISCUSSION THREAD

**Discussion by philipodev** [mantinedev/mantine#4011] [Feature requests]
I've been using `react-final-form` for years, up until 6 months ago when i started using mantine/form.

Currently @mantine/form covers most of the functionality that of final-form.
The only one i really miss is their [submitting](https://final-form.org/docs/final-form/types/FormState#submitting) property.

Currently i have to manage this property in a `useState` or similar.

```tsx
const form = useForm({
  ...options
});

async function onSubmit(){
  // try { fetch etc... 
}

// this is true if the Promise in `onSubmit` is not settled
form.submitting // or form.isSubmitting
```

A very simple fix that allows us to use this property with `LoadingOverlay` or `disabled` props on buttons or inputs (nothing forced)

With your approval i could start working on this and submit a PR.

Cheers 🍻

**Comment by rtivital** [comment]
I'm fine with adding this feature. Although it will require additional fields and methods, not only `submitting`:
- `form.submitting` – indicates current submit status
- `form.setSubmitting(boolean)` – sets submit status
- `form.submitError` – error that was received from `onSubmit` function
- `form.setSubmitError(any)` – function to set submit error
- `form.resetSubmitError()` – function to set submit error to null

**Reply by philipodev** [reply]
Sounds great! Will draft up a PR for this

**Reply by treemmett** [reply]
Any chance this could get merged soon?

https://github.com/mantinedev/mantine/pull/4020

**Comment by treemmett** [comment]
In addition, what are your thoughts on an option to extend `GetInputProps` to return ` { disabled: true }` ?

**Reply by rtivital** [reply]
In which case should it return this?

**Reply by treemmett** [reply]
My use case for the `submitting` property would be to disable form input while the form is posting. Extending `GetInputProps` would prevent the verbosity of adding `disabled={submiting}` to every input in my form.

**Reply by philipodev** [reply]
Im not a fan of forcing people to handle forms. So automatixally setting disabled on all fields while submitting is something that shpuld not be default behaviour. Maybe I can add an extra option to forms `disableFieldsWhileSubmitting`?

**Reply by rtivital** [reply]
No, this is outside of useForm hook scope
